### PR TITLE
Check in/34905/day of skip

### PR DESCRIPTION
**Pull request by brianseek**
## Description
Updated day-of app to use form router for page skipping. Also refactors withLoadedData wrapper to a loading page.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#34905

## Testing done
Updated unit tests and cypress tests.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by timwright12 — APPROVED** [review]

File: src/applications/check-in/utils/navigation/day-of/index.js [hunk 1]
```diff
@@ -31,48 +34,58 @@ const CHECK_IN_FORM_PAGES = Object.freeze([
...
     order: 2,
   },
   {
-    url: URLS.EMERGENCY_CONTACT,
+    url: URLS.NEXT_OF_KIN,
     order: 3,
   },
   {
-    url: URLS.UPDATE_INSURANCE,
+    url: URLS.EMERGENCY_CONTACT,
```
Comment (mdewey) [inline src/applications/check-in/utils/navigation/day-of/index.js:49]: Not part of this PR, but a heads up, that Emergency Contact should come before next of kin

File: src/applications/check-in/day-of/pages/LoadingPage/index.jsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      isEmergencyContactEnabled,
+      checkInExperienceUpdateInformationPageEnabled,
+    ],
+  );
+
+  useEffect(
+    () => {
+      let isCancelled = false;
+      const session = getCurrentToken(window);
+      if (!context || !session) {
```
Comment (mdewey) [inline src/applications/check-in/day-of/pages/LoadingPage/index.jsx:70]: Do we need the session check if we are still using the `withSession`?

File: src/applications/check-in/day-of/pages/LoadingPage/index.jsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        dispatch(receivedMultipleAppointmentDetails(appts, token));
+        dispatch(
+          updateFormAction({
+            patientDemographicsStatus,
+            isEmergencyContactEnabled,
+            checkInExperienceUpdateInformationPageEnabled,
+          }),
+        );
+        if (typeof demo !== 'undefined') {
+          dispatch(receivedDemographicsData(demo));
```
Comment (mdewey) [inline src/applications/check-in/day-of/pages/LoadingPage/index.jsx:49]: Is this now the time to combine the `receievedXXXXX` dispatches?

File: src/applications/check-in/day-of/containers/withLoadedData.jsx [hunk 4]
```diff
@@ -1,134 +0,0 @@
...
-      (payload, token) => {
-        batch(() => {
-          const {
-            appointments: appts,
-            demographics: demo,
-            patientDemographicsStatus,
-          } = payload;
-          dispatch(triggerRefresh(false));
-          dispatch(receivedMultipleAppointmentDetails(appts, token));
-          dispatch(receivedDemographicsStatus(patientDemographicsStatus));
```
Comment (mdewey) [inline src/applications/check-in/day-of/containers/withLoadedData.jsx:48]: Why was this action removed?

**Review by mdewey — CHANGES_REQUESTED** [review]
I like where this is going, just a few questions.

File: src/applications/check-in/day-of/containers/withLoadedData.jsx (hunk 4 above)
Comment (brianseek) [reply src/applications/check-in/day-of/containers/withLoadedData.jsx:48]: I removed this because the form updater is handing everything to do with demographic statuses and we no longer need them to be in the store.

File: src/applications/check-in/utils/navigation/day-of/index.js (hunk 1 above)
Comment (brianseek) [reply src/applications/check-in/utils/navigation/day-of/index.js:49]: good to know, I'm going to rebase and catch up after I get all the tests sorted.

File: src/applications/check-in/day-of/pages/LoadingPage/index.jsx (hunk 2 above)
Comment (brianseek) [reply src/applications/check-in/day-of/pages/LoadingPage/index.jsx:70]: oh good question, I'll look into that. I think I just tried to copy/paste as much of our existing flow as I could.

File: src/applications/check-in/day-of/pages/LoadingPage/index.jsx (hunk 3 above)
Comment (brianseek) [reply src/applications/check-in/day-of/pages/LoadingPage/index.jsx:49]: Yeah, I was thinking about that too. I think Ideally we could have one dispatch for all of the payload kind of like how pre-check-in is doing it. Good stuff for the details on the next ticket.

File: src/applications/check-in/day-of/pages/LoadingPage/index.jsx (hunk 2 above)
Comment (brianseek) [reply src/applications/check-in/day-of/pages/LoadingPage/index.jsx]: removed session check

File: src/applications/check-in/day-of/pages/LoadingPage/index.jsx (hunk 3 above)
Comment (brianseek) [reply src/applications/check-in/day-of/pages/LoadingPage/index.jsx:45]: Add this as a note in the description of https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/34902

File: src/applications/check-in/reducers/index.js [hunk 5]
```diff
@@ -75,8 +75,8 @@ const handler = Object.freeze({
   [TOKEN_WAS_VALIDATED]: tokenWasValidatedHandler,
   [TRIGGER_REFRESH]: triggerRefreshHandler,
   [SEE_STAFF_MESSAGE_UPDATED]: seeStaffMessageUpdatedHandler,
-  [RECEIVED_DEMOGRAPHICS_STATUS]: receivedDemographicsStatusHandler,
-  [UPDATE_FORM]: updateFormHandler,
+  [UPDATE_PRE_CHECK_IN_FORM]: updateFormHandler,
```
Comment (mdewey) [inline src/applications/check-in/reducers/index.js:78]: Just to say it out loud, I think these could probably be combined(?) But that can be on the refactor selectors ticket, not this ticket.

File: src/applications/check-in/day-of/pages/CheckIn/index.jsx [hunk 6]
```diff
@@ -1,29 +1,55 @@
...
+  const { context, appointments } = useSelector(selectAppointmentListData);
+  const appointment = appointments ? appointments[0] : {};
+
   const { token } = context;
   const dispatch = useDispatch();
 
   const getMultipleAppointments = useCallback(
     () => {
-      dispatch(triggerRefresh());
+      let isCancelled = false;
```
Comment (mdewey) [inline src/applications/check-in/day-of/pages/CheckIn/index.jsx:25]: What does `isCancelled` doing?

**Review by mdewey — APPROVED** [review]
Minor tweaks, nothing crazy required, but approved so you can merge before the end of the day

File: src/applications/check-in/reducers/index.js (hunk 5 above)
Comment (brianseek) [reply src/applications/check-in/reducers/index.js:78]: The thing that I couldn't think around was where to handle the different base URLs for the applications. I think once we get closer the unity of the apps there can be a dynamic URLS builder that could solve that problem and allow them to be combined.
